--- pugixml.cpp.orig	2015-02-11 07:48:39.177347007 +0100
+++ pugixml.cpp	2015-02-11 08:06:49.153322687 +0100
@@ -19,7 +19,7 @@
 #ifndef SOURCE_PUGIXML_CPP
 #define SOURCE_PUGIXML_CPP
 
-#include "pugixml.hpp"
+#include "pugixml.h"
 
 #include <stdlib.h>
 #include <stdio.h>
@@ -112,17 +112,17 @@
 #endif
 
 #ifdef PUGIXML_HEADER_ONLY
-#	define PUGI__NS_BEGIN namespace pugi { namespace impl {
-#	define PUGI__NS_END } }
+#	define PUGI__NS_BEGIN namespace ufal { namespace microrestd { namespace pugi { namespace impl {
+#	define PUGI__NS_END } } } }
 #	define PUGI__FN inline
 #	define PUGI__FN_NO_INLINE inline
 #else
 #	if defined(_MSC_VER) && _MSC_VER < 1300 // MSVC6 seems to have an amusing bug with anonymous namespaces inside namespaces
-#		define PUGI__NS_BEGIN namespace pugi { namespace impl {
-#		define PUGI__NS_END } }
+#		define PUGI__NS_BEGIN namespace ufal { namespace microrestd { namespace pugi { namespace impl {
+#		define PUGI__NS_END } } } }
 #	else
-#		define PUGI__NS_BEGIN namespace pugi { namespace impl { namespace {
-#		define PUGI__NS_END } } }
+#		define PUGI__NS_BEGIN namespace ufal { namespace microrestd { namespace pugi { namespace impl { namespace {
+#		define PUGI__NS_END } } } } }
 #	endif
 #	define PUGI__FN
 #	define PUGI__FN_NO_INLINE PUGI__NO_INLINE
@@ -190,9 +190,14 @@
 	{
 		assert(src && dst);
 
+		const char_t* colon; // Ignore XML namespaces, by Milan Straka.
 	#ifdef PUGIXML_WCHAR_MODE
+		if ((colon = wcschr(src, ':'))) src = colon; // Ignore XML namespaces, by Milan Straka.
+		if ((colon = wcschr(dst, ':'))) dst = colon; // Ignore XML namespaces, by Milan Straka.
 		return wcscmp(src, dst) == 0;
 	#else
+		if ((colon = strchr(src, ':'))) src = colon; // Ignore XML namespaces, by Milan Straka.
+		if ((colon = strchr(dst, ':'))) dst = colon; // Ignore XML namespaces, by Milan Straka.
 		return strcmp(src, dst) == 0;
 	#endif
 	}
@@ -207,6 +212,9 @@
 		return lhs[count] == 0;
 	}
 
+// Define the following functions only when using them, by Milan Straka
+#if defined(PUGI__MSVC_CRT_VERSION) || defined(__BORLANDC__) || (defined(__MINGW32__) && (!defined(__STRICT_ANSI__) || defined(__MINGW64_VERSION_MAJOR)))
+#else
 	// Get length of wide string, even if CRT lacks wide character support
 	PUGI__FN size_t strlength_wide(const wchar_t* s)
 	{
@@ -220,6 +228,7 @@
 		return static_cast<size_t>(end - s);
 	#endif
 	}
+#endif
 
 #ifdef PUGIXML_WCHAR_MODE
 	// Convert string to wide string, assuming all symbols are ASCII
@@ -467,6 +476,8 @@
 	}
 PUGI__NS_END
 
+namespace ufal {
+namespace microrestd {
 namespace pugi
 {
 	/// A 'name=value' XML attribute structure.
@@ -510,6 +521,8 @@
 		xml_attribute_struct*	first_attribute;		///< First attribute
 	};
 }
+} // namespace microrestd
+} // namespace ufal
 
 PUGI__NS_BEGIN
 	struct xml_extra_buffer
@@ -1649,6 +1662,9 @@
 	}
 #endif
 
+// Define the following functions only when using them, by Milan Straka
+#if defined(PUGI__MSVC_CRT_VERSION) || defined(__BORLANDC__) || (defined(__MINGW32__) && (!defined(__STRICT_ANSI__) || defined(__MINGW64_VERSION_MAJOR)))
+#else
 	PUGI__FN size_t as_utf8_begin(const wchar_t* str, size_t length)
 	{
 		// get length in utf8 characters
@@ -1667,6 +1683,7 @@
 		// zero-terminate
 		buffer[size] = 0;
 	}
+#endif
 	
 #ifndef PUGIXML_NO_STL
 // Removed the XPATH and STL module, by Milan Straka
@@ -4123,6 +4140,8 @@
 	}
 PUGI__NS_END
 
+namespace ufal {
+namespace microrestd {
 namespace pugi
 {
 	PUGI__FN xml_writer_file::xml_writer_file(void* file_): file(file_)
@@ -5885,6 +5904,8 @@
 		return impl::xml_memory::deallocate;
 	}
 }
+} // namespace microrestd
+} // namespace ufal
 
 #if !defined(PUGIXML_NO_STL) && (defined(_MSC_VER) || defined(__ICC))
 // Removed the XPATH and STL module, by Milan Straka
