--- pugixml.h.orig	2015-02-11 07:48:39.149347008 +0100
+++ pugixml.h	2015-02-11 07:50:58.457343900 +0100
@@ -22,7 +22,7 @@
 #endif
 
 // Include user configuration file (this can define various configuration macros)
-#include "pugiconfig.hpp"
+#include "pugiconfig.h"
 
 #ifndef HEADER_PUGIXML_HPP
 #define HEADER_PUGIXML_HPP
@@ -67,6 +67,7 @@
 #endif
 
 // If the platform is known to have long long support, enable long long functions
+#if 0 // Disabled by Milan Straka because this triggers on MingW, but %lld does not work.
 #ifndef PUGIXML_HAS_LONG_LONG
 #	if defined(__cplusplus) && __cplusplus >= 201103
 #		define PUGIXML_HAS_LONG_LONG
@@ -74,6 +75,7 @@
 #		define PUGIXML_HAS_LONG_LONG
 #	endif
 #endif
+#endif
 
 // Character interface macros
 #ifdef PUGIXML_WCHAR_MODE
@@ -84,6 +86,8 @@
 #	define PUGIXML_CHAR char
 #endif
 
+namespace ufal {
+namespace microrestd {
 namespace pugi
 {
 	// Character type used for all internal storage and operations; depends on PUGIXML_WCHAR_MODE
@@ -93,8 +97,12 @@
 // Removed the XPATH and STL module, by Milan Straka
 #endif
 }
+} // namespace microrestd
+} // namespace ufal
 
 // The PugiXML namespace
+namespace ufal {
+namespace microrestd {
 namespace pugi
 {
 	// Tree node types
@@ -986,6 +994,8 @@
 	allocation_function PUGIXML_FUNCTION get_memory_allocation_function();
 	deallocation_function PUGIXML_FUNCTION get_memory_deallocation_function();
 }
+} // namespace microrestd
+} // namespace ufal
 
 #if !defined(PUGIXML_NO_STL) && (defined(_MSC_VER) || defined(__ICC))
 // Removed the XPATH and STL module, by Milan Straka
