--- daemon.cpp.orig	2015-02-16 08:20:53.456182247 +0100
+++ daemon.cpp	2015-02-16 13:31:34.712453850 +0100
@@ -70,6 +70,10 @@
 #include <windows.h>
 #endif
 
+namespace ufal {
+namespace microrestd {
+namespace libmicrohttpd {
+
 #ifndef HAVE_ACCEPT4
 #define HAVE_ACCEPT4 0
 #endif
@@ -132,7 +136,7 @@
  * @param reason error message with details
  */
 static void
-mhd_panic_std (void *cls,
+mhd_panic_std (void * /*cls*/,
 	       const char *file,
 	       unsigned int line,
 	       const char *reason)
@@ -322,7 +326,7 @@
   if (0 == daemon->per_ip_connection_limit)
     return MHD_YES;
 
-  if (NULL == (key = malloc (sizeof(*key))))
+  if (NULL == (key = (struct MHD_IPCount*) malloc (sizeof(*key))))
     return MHD_NO;
 
   /* Initialize key */
@@ -335,7 +339,7 @@
   MHD_ip_count_lock (daemon);
 
   /* Search for the IP address */
-  if (NULL == (nodep = tsearch (key,
+  if (NULL == (nodep = (void**) tsearch (key,
 				&daemon->per_ip_connection_count,
 				&MHD_ip_addr_compare)))
     {
@@ -391,7 +395,7 @@
   MHD_ip_count_lock (daemon);
 
   /* Search for the IP address */
-  if (NULL == (nodep = tfind (&search_key,
+  if (NULL == (nodep = (void**) tfind (&search_key,
 			      &daemon->per_ip_connection_count,
 			      &MHD_ip_addr_compare)))
     {
@@ -620,7 +624,7 @@
         return MHD_NO;
     }
 #else  // ! MHD_WINSOCK_SOCKETS
-  if (fd >= fd_setsize)
+  if (fd >= int(fd_setsize))
     return MHD_NO;
 #endif // ! MHD_WINSOCK_SOCKETS
   FD_SET (fd, set);
@@ -689,7 +693,7 @@
 MHD_get_fdset2 (struct MHD_Daemon *daemon,
                fd_set *read_fd_set,
                fd_set *write_fd_set,
-               fd_set *except_fd_set,
+               fd_set * /*except_fd_set*/,
                MHD_socket *max_fd,
                unsigned int fd_setsize)
 {
@@ -762,7 +766,7 @@
 static MHD_THRD_RTRN_TYPE_ MHD_THRD_CALL_SPEC_
 MHD_handle_connection (void *data)
 {
-  struct MHD_Connection *con = data;
+  struct MHD_Connection *con = (struct MHD_Connection*) data;
   int num_ready;
   fd_set rs;
   fd_set ws;
@@ -783,7 +787,7 @@
       if (timeout > 0)
 	{
 	  now = MHD_monotonic_time();
-	  if (now - con->last_activity > timeout)
+	  if (unsigned(now - con->last_activity) > timeout)
 	    tv.tv_sec = 0;
 	  else
 	    tv.tv_sec = timeout - (now - con->last_activity);
@@ -962,7 +966,11 @@
       MHD_set_socket_errno_ (ENOTCONN);
       return -1;
     }
+#if WINDOWS
+  ret = recv (connection->socket_fd, (char*) other, i, MSG_NOSIGNAL);
+#else
   ret = recv (connection->socket_fd, other, i, MSG_NOSIGNAL);
+#endif
 #if EPOLL_SUPPORT
   if (ret < (ssize_t) i)
     {
@@ -1001,7 +1009,11 @@
       return -1;
     }
   if (0 != (connection->daemon->options & MHD_USE_SSL))
+#if WINDOWS
+    return send (connection->socket_fd, (const char*) other, i, MSG_NOSIGNAL);
+#else
     return send (connection->socket_fd, other, i, MSG_NOSIGNAL);
+#endif
 #if LINUX
   if ( (connection->write_buffer_append_offset ==
 	connection->write_buffer_send_offset) &&
@@ -1038,7 +1050,11 @@
 	 http://lists.gnu.org/archive/html/libmicrohttpd/2011-02/msg00015.html */
     }
 #endif
+#if WINDOWS
+  ret = send (connection->socket_fd, (const char*) other, i, MSG_NOSIGNAL);
+#else
   ret = send (connection->socket_fd, other, i, MSG_NOSIGNAL);
+#endif
 #if EPOLL_SUPPORT
   if (ret < (ssize_t) i)
     {
@@ -1260,7 +1276,7 @@
 #endif
 #endif
 
-  if (NULL == (connection = malloc (sizeof (struct MHD_Connection))))
+  if (NULL == (connection = (struct MHD_Connection*) malloc (sizeof (struct MHD_Connection))))
     {
       eno = errno;
 #if HAVE_MESSAGES
@@ -1294,7 +1310,7 @@
     }
 
   connection->connection_timeout = daemon->connection_timeout;
-  if (NULL == (connection->addr = malloc (addrlen)))
+  if (NULL == (connection->addr = (sockaddr*) malloc (addrlen)))
     {
       eno = errno;
 #if HAVE_MESSAGES
@@ -1827,6 +1843,7 @@
 #if HAVE_ACCEPT4
   s = accept4 (fd, addr, &addrlen, SOCK_CLOEXEC | nonblock);
 #else
+  nonblock = 0; if (nonblock) s = 0;
   s = accept (fd, addr, &addrlen);
 #endif
   if ((MHD_INVALID_SOCKET == s) || (addrlen <= 0))
@@ -1933,9 +1950,6 @@
 	}
       if (MHD_INVALID_SOCKET != pos->socket_fd)
 	{
-#ifdef WINDOWS
-	  shutdown (pos->socket_fd, SHUT_WR);
-#endif
 	  if (0 != MHD_socket_close_ (pos->socket_fd))
 	    MHD_PANIC ("close failed\n");
 	}
@@ -1999,7 +2013,7 @@
       if (0 != pos->connection_timeout)
 	{
 	  if ( (! have_timeout) ||
-	       (earliest_deadline > pos->last_activity + pos->connection_timeout) )
+	       (earliest_deadline > time_t(pos->last_activity + pos->connection_timeout)) )
 	    earliest_deadline = pos->last_activity + pos->connection_timeout;
 #if HTTPS_SUPPORT
 	  if (  (0 != (daemon->options & MHD_USE_SSL)) &&
@@ -2015,7 +2029,7 @@
        (0 != pos->connection_timeout) )
     {
       if ( (! have_timeout) ||
-	   (earliest_deadline > pos->last_activity + pos->connection_timeout) )
+	   (earliest_deadline > time_t(pos->last_activity + pos->connection_timeout)) )
 	earliest_deadline = pos->last_activity + pos->connection_timeout;
 #if HTTPS_SUPPORT
       if (  (0 != (daemon->options & MHD_USE_SSL)) &&
@@ -2059,7 +2073,7 @@
 MHD_run_from_select (struct MHD_Daemon *daemon,
 		     const fd_set *read_fd_set,
 		     const fd_set *write_fd_set,
-		     const fd_set *except_fd_set)
+		     const fd_set * /*except_fd_set*/)
 {
   MHD_socket ds;
   char tmp;
@@ -2390,10 +2404,12 @@
   unsigned int poll_count;
   int poll_listen;
 
+  bool at_connection_limit = daemon->connections == daemon->connection_limit; // Do not accept when at connection limit, by Milan Straka
+
   memset (&p, 0, sizeof (p));
   poll_count = 0;
   poll_listen = -1;
-  if (MHD_INVALID_SOCKET != daemon->socket_fd)
+  if (!at_connection_limit && MHD_INVALID_SOCKET != daemon->socket_fd)
     {
       p[poll_count].fd = daemon->socket_fd;
       p[poll_count].events = POLLIN;
@@ -2411,7 +2427,7 @@
   if (MHD_NO == may_block)
     timeout = 0;
   else
-    timeout = -1;
+    timeout = at_connection_limit ? 50 : -1;
   if (0 == poll_count)
     return MHD_YES;
   if (poll (p, poll_count, timeout) < 0)
@@ -2443,22 +2459,31 @@
  * @param may_block #MHD_YES if blocking, #MHD_NO if non-blocking
  * @return #MHD_NO on serious errors, #MHD_YES on success
  */
+#ifdef HAVE_POLL_H
+
 static int
 MHD_poll (struct MHD_Daemon *daemon,
 	  int may_block)
 {
-#ifdef HAVE_POLL_H
   if (MHD_YES == daemon->shutdown)
     return MHD_NO;
   if (0 == (daemon->options & MHD_USE_THREAD_PER_CONNECTION))
     return MHD_poll_all (daemon, may_block);
   else
     return MHD_poll_listen_socket (daemon, may_block);
+}
+
 #else
+
+static int
+MHD_poll (struct MHD_Daemon * /*daemon*/,
+	  int /*may_block*/)
+{
   return MHD_NO;
-#endif
 }
 
+#endif
+
 
 #if EPOLL_SUPPORT
 
@@ -2730,7 +2755,7 @@
 static MHD_THRD_RTRN_TYPE_ MHD_THRD_CALL_SPEC_
 MHD_select_thread (void *cls)
 {
-  struct MHD_Daemon *daemon = cls;
+  struct MHD_Daemon *daemon = (struct MHD_Daemon*) cls;
 
   while (MHD_YES != daemon->shutdown)
     {
@@ -2760,8 +2785,8 @@
  *  shorter afterwards due to elimination of escape sequences)
  */
 static size_t
-unescape_wrapper (void *cls,
-                  struct MHD_Connection *connection,
+unescape_wrapper (void * /*cls*/,
+                  struct MHD_Connection * /*connection*/,
                   char *val)
 {
   return MHD_http_unescape (val);
@@ -2979,7 +3004,7 @@
           break;
         case MHD_OPTION_THREAD_POOL_SIZE:
           daemon->worker_pool_size = va_arg (ap, unsigned int);
-	  if (daemon->worker_pool_size >= (SIZE_MAX / sizeof (struct MHD_Daemon)))
+	  if (daemon->worker_pool_size >= (UINT_MAX / sizeof (struct MHD_Daemon)))
 	    {
 #if HAVE_MESSAGES
 	      MHD_DLOG (daemon,
@@ -3404,7 +3429,7 @@
 #endif
   if (NULL == dh)
     return NULL;
-  if (NULL == (daemon = malloc (sizeof (struct MHD_Daemon))))
+  if (NULL == (daemon = (struct MHD_Daemon*) malloc (sizeof (struct MHD_Daemon))))
     return NULL;
   memset (daemon, 0, sizeof (struct MHD_Daemon));
 #if EPOLL_SUPPORT
@@ -3422,11 +3447,6 @@
   daemon->socket_fd = MHD_INVALID_SOCKET;
   daemon->listening_address_reuse = 0;
   daemon->options = (enum MHD_OPTION) flags;
-#if WINDOWS
-  /* Winsock is broken with respect to 'shutdown';
-     this disables us calling 'shutdown' on W32. */
-  daemon->options |= MHD_USE_EPOLL_TURBO;
-#endif
   daemon->port = port;
   daemon->apc = apc;
   daemon->apc_cls = apc_cls;
@@ -3518,7 +3538,7 @@
 	  free (daemon);
 	  return NULL;
 	}
-      daemon->nnc = malloc (daemon->nonce_nc_size * sizeof (struct MHD_NonceNc));
+      daemon->nnc = (struct MHD_NonceNc*) malloc (daemon->nonce_nc_size * sizeof (struct MHD_NonceNc));
       if (NULL == daemon->nnc)
 	{
 #if HAVE_MESSAGES
@@ -3610,7 +3630,11 @@
           if (0 > setsockopt (socket_fd,
                               SOL_SOCKET,
                               SO_REUSEADDR,
+#if WINDOWS
+                              (const char*)&on, sizeof (on)))
+#else
                               (void*)&on, sizeof (on)))
+#endif
           {
 #if HAVE_MESSAGES
             MHD_DLOG (daemon,
@@ -3631,7 +3655,7 @@
           if (0 > setsockopt (socket_fd,
                               SOL_SOCKET,
                               SO_REUSEADDR,
-                              (void*)&on, sizeof (on)))
+                              (const char*)&on, sizeof (on)))
             {
 #if HAVE_MESSAGES
               MHD_DLOG (daemon,
@@ -3684,7 +3708,7 @@
           if (0 > setsockopt (socket_fd,
                               SOL_SOCKET,
                               SO_EXCLUSIVEADDRUSE,
-                              (void*)&on, sizeof (on)))
+                              (const char*)&on, sizeof (on)))
             {
 #if HAVE_MESSAGES
               MHD_DLOG (daemon,
@@ -3811,7 +3835,7 @@
 	    }
 	}
 #endif
-      if (listen (socket_fd, 32) < 0)
+      if (listen (socket_fd, 128) < 0) // Increased to 128 from 32 by Milan Straka.
 	{
 #if HAVE_MESSAGES
           MHD_DLOG (daemon,
@@ -3964,7 +3988,7 @@
 #endif /* WINDOWS && !CYGWIN */
 
       /* Allocate memory for pooled objects */
-      daemon->worker_pool = malloc (sizeof (struct MHD_Daemon)
+      daemon->worker_pool = (struct MHD_Daemon*) malloc (sizeof (struct MHD_Daemon)
                                     * daemon->worker_pool_size);
       if (NULL == daemon->worker_pool)
         goto thread_failed;
@@ -4630,5 +4654,9 @@
 
 _SET_INIT_AND_DEINIT_FUNCS(MHD_init, MHD_fini);
 
+} // namespace libmicrohttpd
+} // namespace microrestd
+} // namespace ufal
+
 /* end of daemon.c */
 
