--- w32functions.h.orig	2015-02-10 15:42:42.075316563 +0100
+++ w32functions.h	2015-02-10 15:44:55.855320247 +0100
@@ -39,10 +39,9 @@
 #include <winsock2.h>
 #include "platform_interface.h"
 
-#ifdef __cplusplus
-extern "C"
-{
-#endif
+namespace ufal {
+namespace microrestd {
+namespace libmicrohttpd {
 
 #define MHDW32ERRBASE 3300
 
@@ -199,7 +198,7 @@
 /* Emulate snprintf function on W32 */
 int W32_snprintf(char *__restrict s, size_t n, const char *__restrict format, ...);
 
-#ifdef __cplusplus
-}
-#endif
+} // namespace libmicrohttpd
+} // namespace microrestd
+} // namespace ufal
 #endif //MHD_W32FUNCTIONS_H
